### ISSUE TITLE
Counted queries and change streams

### ISSUE THREAD

**Issue by aravindet** [usegraffy/graffy#7]
TL:DR; Some watch() providers may handle `{ after: '', before: 'b' }` but not `{ first: 15 }`. How do they comunicate this?

----

### Original write-up

Graffy providers often have limitations around what queries they can fulfil. They need to be able to signal these limitations, so graffy-fill can figure out ways to work around them.

Currently, we use some ad-hoc mechanisms to signal limitations. Perhaps we could design these in a more systematic way.

### Dangling links

Consider the `posts` and `users` example. Let's say the posts resolver cannot fetch user data - if `author` info was requested, it ignores the nested fields and simply returns a link as the `author` field.

Graffy-fill makes a new (live) query for the linked data.

### Change streams

Imagine a subscription provider that can provide change streams but not the initial result (current state). It signals this by yielding `undefined` as the first value.

Graffy-fill makes a separate fetch to get the initial value.

### Page bounds

Imagine a change stream provider pushing updates for `users`. Say it does not have access to the current state, but can access an event stream of user updates where each update specifies the `user_id`.

Say the query is for the `first 30` users.

In a scenario where there are thousands of users, MOST user updates will be irrelevant for this query. However, there is no way for this provider to know that, because it cannot know the range of IDs that match "first 30".

Perhaps there should be a way for the provider to signal that it cannot serve "counted" pages (i.e. that use first / last parameters) but can serve "bounded" ones (i.e. those that only have before AND after, but no first / last).

Graffy fill could use the fetch results to convert a "counted" page into a "bounded" one. 

NOTE: If the pagination happens in an "index" (nodes where all the children are links), it will work fine if the change stream provider ignored the bounds queries and just pretend like there are no updates. However it seems like this is just working "by accident".

**Comment by aravindet** [comment]
Option: Custom Errors.

`RangesNotSupportedError`, `CountedRangesNotSupportedError`, etc.
